app/vmui/packages/vmui/src/hooks/useSortedCategories.ts

```diff
@@ -23,6 +23,6 @@ export const getColumns = (data: MetricBase[]): MetricCategory[] => {
 export const useSortedCategories = (data: MetricBase[], displayColumns?: string[]): MetricCategory[] => (
   useMemo(() => {
     const sortedColumns = getColumns(data);
-    return displayColumns ? sortedColumns.filter(col => displayColumns.includes(col.key)) : sortedColumns;
+    return displayColumns ? sortedColumns.filter(col => displayColumns.includes(col.key)) : [];
```

Comment:
As for me it is better to make some calculations and return results instead of returning ternary operator. I suggest changes like

```javascript
const filteredCols = sortedColumns.filter(col => displayColumns.includes(col.key));
const foundCols = displayColumns ? filteredCols : [];
return foundCols;
``` 
It would be more clear

Reply:
thanks for your review，bro.

Reply:
however, bro , you alse should check displayColumns is null or not. i think it's better use ternary operator.
![image](https://github.com/user-attachments/assets/5cfc3854-2a42-4df6-a4b1-d0e11c541c75)

Reply:
> however, bro , you alse should check displayColumns is null or not. i think it's better use ternary operator. ![image](https://private-user-images.githubusercontent.com/24344673/351256806-5cfc3854-2a42-4df6-a4b1-d0e11c541c75.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3MjU4MjksIm5iZiI6MTcyMTcyNTUyOSwicGF0aCI6Ii8yNDM0NDY3My8zNTEyNTY4MDYtNWNmYzM4NTQtMmE0Mi00ZGY2LWE0YjEtZDBlMTFjNTQxYzc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIzVDA5MDUyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0YWUzN2VlNDA4YmE2NjM5MGUwODFmYWVkYTExMGNmYzdlNDkyMDU3YTlmYmRiODRjNGFmOTA1NjE4ZTg1ZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Qz9IlT4GPrBOummLaPZ9g0TDJk3OeFdACZi-WQFAzd0)

YOu can check the state of that param before the whole logic
somethisng like this

```javascript
export const useSortedCategories = (data: MetricBase[], displayColumns?: string[]): MetricCategory[] => (
  useMemo(() => {
   if (!displayColumns) return [];
    const sortedColumns = getColumns(data);
    return sortedColumns.filter(col => displayColumns.includes(col.key));
  }, [data, displayColumns])
);
```
it is more clear

Reply:
ok, thanks